experimental/Schemes/IdealSheaves.jl

```diff
@@ -0,0 +1,325 @@
+end
+
+# this constructs the empty ideal sheaf
+function IdealSheaf(X::CoveredScheme) 
+  C = default_covering(X)
+  I = IdDict{AbsSpec, Ideal}()
+  for U in basic_patches(C)
+    I[U] = ideal(OO(U), elem_type(OO(U))[])
+  end
+  return IdealSheaf(X, I, check=false)
+end
+
+# internal routine to set up an ideal sheaf by automatic extension 
+# from one prescribed set of generators on one affine patch
+function IdealSheaf(X::CoveredScheme, U::AbsSpec, g::Vector{RET}) where {RET<:RingElem}
+  C = default_covering(X)
+  U in patches(C) || error("the affine open patch does not belong to the covering")
+  for f in g
+    parent(f) === OO(U) || error("the generators do not belong to the correct ring")
+  end
+  D = IdDict{AbsSpec, Ideal}()
+  D[U] = ideal(OO(U), g)
+  D = extend!(C, D)
+  I = IdealSheaf(X, D, check=false)
+  return I
+end
+
+# pullback of an ideal sheaf for internal use between coverings of the same scheme
+#function (F::CoveringMorphism)(I::IdealSheaf)
+#  X = scheme(I)
+#  D = codomain(F)
+#  D == covering(I) || error("ideal sheaf is not defined on the correct covering")
+#  C = domain(F)
+#  new_dict = Dict{AbsSpec, Ideal}()
+#
+#  # go through the patches of C and pull back the generators 
+#  # whenever they are defined on the target patch
+#  for U in patches(C)
+#    f = F[U]
+#    V = codomain(f)
+#    # for the basic patches here
+#    if haskey(ideal_dict(I), V)
+#      new_dict[U] = ideal(OO(U), pullback(f).(I[V]))
+#    end
+#    # check for affine refinements
+#    if haskey(affine_refinements(D), V)
+#      Vrefs = affine_refinements(D)[V]
+#      # pull back the refinement
+#      for W in Vrefs
+#        h = pullback(f).(gens(W))
+#        # take care to discard possibly empty preimages of patches
+#        j = [i for i in 1:length(h) if !iszero(h)]
+#        Wpre = SpecOpen(U, h[j])
+#        add_affine_refinement!(C, Wpre)
+#        for i in 1:length(j)
+#          if haskey(ideal_dict(I), Wpre[i])
+#            new_dict[Wpre[i]] = lifted_numerator.(pullback(f).(I[V[j[i]]]))
+#          end
+#        end
+#      end
+#    end
+#  end
+#  return IdealSheaf(X, C, new_dict)
+#end
+
+function +(I::IdealSheaf, J::IdealSheaf) 
+  X = space(I)
+  X == space(J) || error("ideal sheaves are not defined over the same scheme")
+  new_dict = IdDict{AbsSpec, Ideal}()
+  CI = default_covering(X)
+  for U in patches(CI)
+    new_dict[U] = I(U) + J(U)
+  end
+  return IdealSheaf(X, new_dict, check=false)
+end
+
+function *(I::IdealSheaf, J::IdealSheaf) 
+  X = space(I)
+  X == space(J) || error("ideal sheaves are not defined over the same scheme")
+  new_dict = IdDict{AbsSpec, Ideal}()
+  CI = default_covering(X)
+  for U in patches(CI)
+    new_dict[U] = I(U) * J(U)
+  end
+  return IdealSheaf(X, new_dict, check=false)
+end
+
+@Markdown.doc """
+    simplify!(I::IdealSheaf)
+
+Replaces the set of generators of the ideal sheaf by a minimal 
+set of random linear combinations in every affine patch. 
```

Comment:
Could you please explain why this is desirable/better?

Reply:
Less generators are less of a mess. 

No, seriously: The point is that one of the things you would like to do with an ideal sheaf is to blow them up. That means introducing an extra variable for every generator, so you really want to keep this number small. There will be more sophisticated methods, e.g. to produce local complete intersections. But this is a first simple one.